File: sdks/java/src/main/java/com/boltprediction/sdk/model/DegradationParamsSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 预紧力劣化模型参数 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class DegradationParamsSchema {

    @JsonProperty("nominal_preload")
    private Double nominalPreload;

    @JsonProperty("min_effective_preload_ratio")
    private Double minEffectivePreloadRatio;

    @JsonProperty("relaxation_rate_per_month")
    private Double relaxationRatePerMonth;

    @JsonProperty("temperature_acceleration_factor")
    private Double temperatureAccelerationFactor;

    @JsonProperty("vibration_acceleration_factor")
    private Double vibrationAccelerationFactor;

    @JsonProperty("cycle_acceleration_factor")
    private Double cycleAccelerationFactor;

    public Double getNominalPreload() {
        return nominalPreload;
    }

    public void setNominalPreload(Double nominalPreload) {
        this.nominalPreload = nominalPreload;
    }

    public Double getMinEffectivePreloadRatio() {
        return minEffectivePreloadRatio;
    }

    public void setMinEffectivePreloadRatio(Double minEffectivePreloadRatio) {
        this.minEffectivePreloadRatio = minEffectivePreloadRatio;
    }

    public Double getRelaxationRatePerMonth() {
        return relaxationRatePerMonth;
    }

    public void setRelaxationRatePerMonth(Double relaxationRatePerMonth) {
        this.relaxationRatePerMonth = relaxationRatePerMonth;
    }

    public Double getTemperatureAccelerationFactor() {
        return temperatureAccelerationFactor;
    }

    public void setTemperatureAccelerationFactor(Double temperatureAccelerationFactor) {
        this.temperatureAccelerationFactor = temperatureAccelerationFactor;
    }

    public Double getVibrationAccelerationFactor() {
        return vibrationAccelerationFactor;
    }

    public void setVibrationAccelerationFactor(Double vibrationAccelerationFactor) {
        this.vibrationAccelerationFactor = vibrationAccelerationFactor;
    }

    public Double getCycleAccelerationFactor() {
        return cycleAccelerationFactor;
    }

    public void setCycleAccelerationFactor(Double cycleAccelerationFactor) {
        this.cycleAccelerationFactor = cycleAccelerationFactor;
    }

}

File: sdks/python/bolt_prediction_sdk/models/degradation_params_schema.py
"""
DegradationParamsSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class DegradationParamsSchema(SDKBaseModel):
    """预紧力劣化模型参数"""

    nominal_preload: Optional[float] = Field(description="额定预紧力 (kN)", default=600.0)
    min_effective_preload_ratio: Optional[float] = Field(description="最小有效压紧比阈值", default=0.6)
    relaxation_rate_per_month: Optional[float] = Field(description="自然松弛月速率", default=0.015)
    temperature_acceleration_factor: Optional[float] = Field(description="高温加速因子 (每°C高于40)", default=0.002)
    vibration_acceleration_factor: Optional[float] = Field(description="振动加速因子", default=0.003)
    cycle_acceleration_factor: Optional[float] = Field(description="压力循环加速因子", default=0.0001)
